### PR TITLE
Classify Alzheimer’s disease(46475781)

### PR DESCRIPTION
**Pull request by ENGGP**
Task: Classify Alzheimer’s disease (normal and AD) of the ADNI brain data (see Appendix for link) using one
of the latest vision transformers such as the GFNet [6] set having a minimum accuracy of 0.8 on the test
set. [Hard Difficulty]

Reference: https://github.com/raoyongming/GFNet

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Wrong file path. All student-constructed documents should be packed in the ‘PatternAnalysis-2024/recognition/<student_constructed_folder_name>/’ folder. Please reorganize your files, otherwise it won't be merged into Shakes' branch.

**Problem Solving:**

- The algorithm cannot solve the problem, as the code cannot run. 
- Accuracy in testing dataset is announced as 0,9388, but no evidence supports this result.

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions. However, the model is initialized as input channel = 3 while the input image channel is 1, so an input shape error will be posted when the images are fed into the model.
- No data augmentation.

**Code design:** Confused.
- Redundant imported libraries, including double imports, and useless imports.
- In dataset.py, all of the code other than import libraries are the executive lines instead of data processing functions though it can be imported by other scripts. in addition, model initializing shouldn't appear in the data pre-processing module.
- The training process is organized in the predict.py script instead of the train.py one.

**Code comment and docstring:**

- Good code comments 
- Minimal function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Minimal commits, just copy-and-paste code.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request shows no incorporation of previous feedback .
Should add  a description about the  corrected File Structure .

File Organizing: Wrong file path. All student-constructed documents should be packed in the ‘PatternAnalysis-2024/recognition/<student_constructed_folder_name>/’ folder. Please reorganize your files, otherwise it won't be merged into Shakes' branch.

Commit Log:
Commit messages are  not meaningful .
Commits do not  demonstrate a logical development flow.
Commits are not  regularly made.

Documentation:
The README file is incomplete .

**Comment by hanemma7moud** [comment]
Mengqi Yu

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Poor design and implementation. </td><td>-2</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Poor solution to problem. The algorithm cannot solve the problem, as the code cannot work and model is not GFNet. Low Acc.</td><td>-4</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure NOT present. Remove unwanted files.</td><td>-1</td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Minimal.</td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. Intermediate.</td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Incomplete.</td><td>-3</td></tr>
	<tr><td></td><td>Model/technical explanation :Missing. </td><td>-3</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Pull Request has problems: Wrong branch.</td><td>-2</td></tr>
	<tr><td></td><td>Feedback action required: Feedback marks possible +2 if the requested changes are made. Remove cahce files.</td><td>-2</td></tr>
	<tr><td></td><td>Request Description is adequate. Minimal.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-24</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Mengqi Yu

**Comment by shakes76** [comment]
No working model incomplete implementation -1 additionally. No results/plots visualised -2. Data splits/leakage -2

Cant merge because of conflicting changes to main repo files (README). Please update for merge, doesn't affect grade.
